lua/rest-nvim/request/init.lua

```diff
@@ -67,23 +66,6 @@ local function get_body(bufnr, start_line, stop_line, has_json)
     end
   end
 
-  local is_json, json_body = pcall(vim.fn.json_decode, body)
-
-  if is_json then
```

Comment:
Don't delete these lines; this statement does two things:
1. LINE 75: Minify json. ie. remove extra tabs and spaces to reduce bandwidth.
2. LINE 78-83: Parse nested json into valid `application/x-www-form-urlencoded` string values. see #138

Reply:
I think 1 is non-relevant, also it means we dont send the exact same payload as specified by the user which is concerning (encoding/decoding floats can raise issues I suppose, same for numbers ?).

As for 2, application/x-www-form-urlencoded seems like a pain https://stackoverflow.com/questions/9870523/what-are-the-differences-between-application-json-and-application-x-www-form-url but indeed that's the curl default. One solution could be to default to application/json but that can be surprising too.

Maybe we could do is implement the same mechanism as for "formatter", but for inputs instead of outputs. have the ability to to preprocess the payload depending on content-type.